assembly/src/ast/imports.rs

```diff
@@ -0,0 +1,91 @@
+use super::{
+    BTreeMap, ByteReader, ByteWriter, Deserializable, DeserializationError, ImportedProcsMap,
+    LibraryPath, ProcedureId, ProcedureName, Serializable, String, ToString, Vec,
+};
+
+/// Information about imports stored in the AST
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct ModuleImports {
+    /// Imported libraries
+    imports: BTreeMap<String, LibraryPath>,
+    /// Imported procedures that are called from somewhere in the AST
+    used_imported_procs: ImportedProcsMap,
+}
+
+impl ModuleImports {
+    pub fn new(
+        imports: BTreeMap<String, LibraryPath>,
+        used_imported_procs: ImportedProcsMap,
+    ) -> Self {
+        Self {
+            imports,
+            used_imported_procs,
+        }
+    }
+
+    pub fn get_module_path(&self, module_name: String) -> Option<&LibraryPath> {
+        self.imports.get(&module_name)
+    }
+
+    pub fn imports_len(&self) -> usize {
+        self.imports.len()
+    }
```

Comment:
If we move validation logic into this struct, we probably can get rid of this method.